frappe/core/doctype/user/user.py

```diff
@@ -1107,3 +1108,15 @@ def generate_keys(user):
 
 		return {"api_secret": api_secret}
 	frappe.throw(frappe._("Not Permitted"), frappe.PermissionError)
+
+
+def expire_reset_password_key():
+	frappe.db.sql("""
+UPDATE `tabUser`
+SET
+	reset_password_key=""
+WHERE
+	datediff(hh, reset_key_created_on, now()) >= 2
```

Comment:
@Thunderbottom `datediff()` is not compatible with postgres.

& 

fix code formatting